[macOS] The color CSS property value is ignored on <select> elements

pxlcoder opened the pull request:
#### d31184adaa0bc64aa4aa523dafd5e0437094d20e
<pre>
[macOS] The color CSS property value is ignored on &lt;select&gt; elements
<a href="https://bugs.webkit.org/show_bug.cgi?id=238667">https://bugs.webkit.org/show_bug.cgi?id=238667</a>
rdar://91167635

Reviewed by Tim Nguyen.

Currently, `appearance: auto` on &lt;select&gt; elements forces a text color,
preventing authors from customizing it. This behavior is different from Chrome
and Firefox, and is a web compatibility issue.

To fix, remove the logic that forces a text color when using &lt;select&gt; elements
with a native appearance. The disabled text color behavior is preserved by an
existing rule in the UA style sheet.

This change also has the effect of using CanvasText rather than ButtonText as
the default text color, which matches pop-up buttons in AppKit and other
text content in WebKit.

* LayoutTests/TestExpectations:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-bigsur/fast/forms/control-restrict-line-height-expected.txt:
* LayoutTests/platform/mac-bigsur/fast/forms/select/optgroup-rendering-expected.txt:
* LayoutTests/platform/mac-bigsur/tables/mozilla/bugs/bug2479-3-expected.txt:
* LayoutTests/platform/mac/editing/pasteboard/4641033-expected.txt:
* LayoutTests/platform/mac/editing/pasteboard/4944770-1-expected.txt:
* LayoutTests/platform/mac/editing/pasteboard/4944770-2-expected.txt:
* LayoutTests/platform/mac/editing/selection/caret-before-select-expected.txt:
* LayoutTests/platform/mac/editing/selection/replaced-boundaries-3-expected.txt:
* LayoutTests/platform/mac/editing/selection/select-box-expected.txt:
* LayoutTests/platform/mac/editing/selection/select-element-paragraph-boundary-expected.txt:
* LayoutTests/platform/mac/fast/block/float/float-avoidance-expected.txt:
* LayoutTests/platform/mac/fast/block/margin-collapse/103-expected.txt:
* LayoutTests/platform/mac/fast/css/text-transform-select-expected.txt:
* LayoutTests/platform/mac/fast/forms/003-expected.txt:
* LayoutTests/platform/mac/fast/forms/004-expected.txt:
* LayoutTests/platform/mac/fast/forms/basic-selects-expected.txt:
* LayoutTests/platform/mac/fast/forms/control-clip-overflow-expected.txt:
* LayoutTests/platform/mac/fast/forms/control-restrict-line-height-expected.txt:
* LayoutTests/platform/mac/fast/forms/disabled-select-change-index-expected.txt:
* LayoutTests/platform/mac/fast/forms/form-element-geometry-expected.txt:
* LayoutTests/platform/mac/fast/forms/menulist-deselect-update-expected.txt:
* LayoutTests/platform/mac/fast/forms/menulist-narrow-width-expected.txt:
* LayoutTests/platform/mac/fast/forms/menulist-no-overflow-expected.txt:
* LayoutTests/platform/mac/fast/forms/menulist-style-color-expected.txt:
* LayoutTests/platform/mac/fast/forms/menulist-width-change-expected.txt:
* LayoutTests/platform/mac/fast/forms/option-script-expected.txt:
* LayoutTests/platform/mac/fast/forms/option-strip-whitespace-expected.txt:
* LayoutTests/platform/mac/fast/forms/option-text-clip-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-align-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-baseline-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-change-listbox-to-popup-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-dirty-parent-pref-widths-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-disabled-appearance-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-element-focus-ring-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-initial-position-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-selected-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-style-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-visual-hebrew-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-writing-direction-natural-expected.txt:
* LayoutTests/platform/mac/fast/forms/select/optgroup-rendering-expected.txt:
* LayoutTests/platform/mac/fast/forms/selectlist-minsize-expected.txt:
* LayoutTests/platform/mac/fast/forms/stuff-on-my-optgroup-expected.txt:
* LayoutTests/platform/mac/fast/invalid/014-expected.txt:
* LayoutTests/platform/mac/fast/parser/document-write-option-expected.txt:
* LayoutTests/platform/mac/fast/replaced/replaced-breaking-expected.txt:
* LayoutTests/platform/mac/fast/replaced/replaced-breaking-mixture-expected.txt:
* LayoutTests/platform/mac/fast/replaced/three-selects-break-expected.txt:
* LayoutTests/platform/mac/fast/replaced/width100percent-menulist-expected.txt:
* LayoutTests/platform/mac/fast/text/international/bidi-menulist-expected.txt:
* LayoutTests/platform/mac/fast/text/international/pop-up-button-text-alignment-and-direction-expected.txt:
* LayoutTests/platform/mac/http/tests/navigation/javascriptlink-frames-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug1188-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug18359-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug2479-3-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug2479-4-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug29326-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug33855-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug4382-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug96334-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/core/margins-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/dom/tableDom-expected.txt:
* LayoutTests/platform/mac/tables/mozilla_expected_failures/bugs/bug2479-5-expected.txt:
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::adjustMenuListStyle const):

Canonical link: <a href="https://commits.webkit.org/253074@main">https://commits.webkit.org/253074@main</a>
</pre>

nt1m reviewed (approved):
* tables/mozilla/bugs/bug2479-3.html needs rebaselining for mac
* imported/w3c/web-platform-tests/css/css-contain/contain-size-select-elem-001.html needs an ImageOnlyFailure expectation for GTK

r=me once EWS is fully green

pxlcoder committed:
[macOS] The color CSS property value is ignored on <select> elements
https://bugs.webkit.org/show_bug.cgi?id=238667
rdar://91167635

Reviewed by Tim Nguyen.

Currently, `appearance: auto` on <select> elements forces a text color,
preventing authors from customizing it. This behavior is different from Chrome
and Firefox, and is a web compatibility issue.

To fix, remove the logic that forces a text color when using <select> elements
with a native appearance. The disabled text color behavior is preserved by an
existing rule in the UA style sheet.

This change also has the effect of using CanvasText rather than ButtonText as
the default text color, which matches pop-up buttons in AppKit and other
text content in WebKit.

* LayoutTests/TestExpectations:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-bigsur/fast/forms/control-restrict-line-height-expected.txt:
* LayoutTests/platform/mac-bigsur/fast/forms/select/optgroup-rendering-expected.txt:
* LayoutTests/platform/mac-bigsur/tables/mozilla/bugs/bug2479-3-expected.txt:
* LayoutTests/platform/mac/editing/pasteboard/4641033-expected.txt:
* LayoutTests/platform/mac/editing/pasteboard/4944770-1-expected.txt:
* LayoutTests/platform/mac/editing/pasteboard/4944770-2-expected.txt:
* LayoutTests/platform/mac/editing/selection/caret-before-select-expected.txt:
* LayoutTests/platform/mac/editing/selection/replaced-boundaries-3-expected.txt:
* LayoutTests/platform/mac/editing/selection/select-box-expected.txt:
* LayoutTests/platform/mac/editing/selection/select-element-paragraph-boundary-expected.txt:
* LayoutTests/platform/mac/fast/block/float/float-avoidance-expected.txt:
* LayoutTests/platform/mac/fast/block/margin-collapse/103-expected.txt:
* LayoutTests/platform/mac/fast/css/text-transform-select-expected.txt:
* LayoutTests/platform/mac/fast/forms/003-expected.txt:
* LayoutTests/platform/mac/fast/forms/004-expected.txt:
* LayoutTests/platform/mac/fast/forms/basic-selects-expected.txt:
* LayoutTests/platform/mac/fast/forms/control-clip-overflow-expected.txt:
* LayoutTests/platform/mac/fast/forms/control-restrict-line-height-expected.txt:
* LayoutTests/platform/mac/fast/forms/disabled-select-change-index-expected.txt:
* LayoutTests/platform/mac/fast/forms/form-element-geometry-expected.txt:
* LayoutTests/platform/mac/fast/forms/menulist-deselect-update-expected.txt:
* LayoutTests/platform/mac/fast/forms/menulist-narrow-width-expected.txt:
* LayoutTests/platform/mac/fast/forms/menulist-no-overflow-expected.txt:
* LayoutTests/platform/mac/fast/forms/menulist-style-color-expected.txt:
* LayoutTests/platform/mac/fast/forms/menulist-width-change-expected.txt:
* LayoutTests/platform/mac/fast/forms/option-script-expected.txt:
* LayoutTests/platform/mac/fast/forms/option-strip-whitespace-expected.txt:
* LayoutTests/platform/mac/fast/forms/option-text-clip-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-align-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-baseline-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-change-listbox-to-popup-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-dirty-parent-pref-widths-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-disabled-appearance-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-element-focus-ring-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-initial-position-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-selected-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-style-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-visual-hebrew-expected.txt:
* LayoutTests/platform/mac/fast/forms/select-writing-direction-natural-expected.txt:
* LayoutTests/platform/mac/fast/forms/select/optgroup-rendering-expected.txt:
* LayoutTests/platform/mac/fast/forms/selectlist-minsize-expected.txt:
* LayoutTests/platform/mac/fast/forms/stuff-on-my-optgroup-expected.txt:
* LayoutTests/platform/mac/fast/invalid/014-expected.txt:
* LayoutTests/platform/mac/fast/parser/document-write-option-expected.txt:
* LayoutTests/platform/mac/fast/replaced/replaced-breaking-expected.txt:
* LayoutTests/platform/mac/fast/replaced/replaced-breaking-mixture-expected.txt:
* LayoutTests/platform/mac/fast/replaced/three-selects-break-expected.txt:
* LayoutTests/platform/mac/fast/replaced/width100percent-menulist-expected.txt:
* LayoutTests/platform/mac/fast/text/international/bidi-menulist-expected.txt:
* LayoutTests/platform/mac/fast/text/international/pop-up-button-text-alignment-and-direction-expected.txt:
* LayoutTests/platform/mac/http/tests/navigation/javascriptlink-frames-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug1188-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug18359-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug2479-3-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug2479-4-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug29326-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug33855-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug4382-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/bugs/bug96334-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/core/margins-expected.txt:
* LayoutTests/platform/mac/tables/mozilla/dom/tableDom-expected.txt:
* LayoutTests/platform/mac/tables/mozilla_expected_failures/bugs/bug2479-5-expected.txt:
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::adjustMenuListStyle const):

Canonical link: https://commits.webkit.org/253074@main
